src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -17,19 +17,52 @@
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
+     * @deprecated since Symfony 6.3
+     *
      * @var string[]
      */
     private $missingArguments;
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    private ?string $class;
+    private ?string $missingArgument;
+
+    public function __construct(string $message, int $code = 0, \Throwable $previous = null, /* string */ $class = null, string $missingArgument = null)
     {
-        $this->missingArguments = $missingArguments;
+        if (\is_array($class)) {
+            trigger_deprecation('symfony/serializer', '6.3', 'Passing an array of missing arguments as the fourth argument of the "%s" constructor is deprecated, pass the missing arguments class name instead and the missing arguments name as the fifth argument.', __CLASS__);
+
+            $this->missingArguments = $class;
+
+            $class = null;
+            $missingArgument = $this->missingArguments[0] ?? null;
+        }
+
+        if (null !== $class && !\is_string($class)) {
```

Comment:
With the `@param` tag, it'll be no need to check it, I guess.

Reply:
The `@param` tag doesn't really validate the right type is passed to the constructor. This code is only here to emulate the behavior that the `string` type hint on the `$class` param would have and should be removed once the type hint is added.